openstack/blockstorage/extensions/volumeactions/requests.go

```diff
@@ -343,3 +343,42 @@ func SetBootable(client *gophercloud.ServiceClient, id string, opts BootableOpts
 	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
 	return
 }
+
+// ChangeTypeOptsBuilder allows extensions to add additional parameters to the
+// ChangeType request.
+type ChangeTypeOptsBuilder interface {
+	ToVolumeChangeTypeMap() (map[string]interface{}, error)
+}
+
+// ChangeTypeOpts contains options for changing the type of an existing Volume.
+// This object is passed to the volumes.ChangeType function.
+type ChangeTypeOpts struct {
+	// NewType is the name of the new volume type of the volume.
+	NewType string `json:"new_type" required:"true"`
+
+	// MigrationPolicy specifies if the volume should be migrated when it is
+	// re-typed. Possible values are "on-demand" or "never". If not specified,
+	// the default is "never".
+	MigrationPolicy string `json:"migration_policy,omitempty"`
```

Comment:
It looks like there are only two possible values for `MigrationPolicy`: https://github.com/openstack/cinder/blob/876ac4e79fe14887ee3902ef833be5e2273d800b/cinder/api/schemas/volume_actions.py#L94-L96

Using a generic `string` is fine, but it's also possible to create a specific type for this like how it's done here:

* https://github.com/gophercloud/gophercloud/blob/master/openstack/networking/v2/extensions/lbaas_v2/listeners/requests.go#L8-L17
* https://github.com/gophercloud/gophercloud/blob/master/openstack/networking/v2/extensions/lbaas_v2/listeners/requests.go#L85

If the migration policy choices rarely change, creating a type is usually a good method for more specific data modeling.

Reply:
Done :)